hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/XceiverClientRatis.java

```diff
@@ -275,39 +302,45 @@ public XceiverClientReply watchForCommit(long index)
     }
 
     try {
-      CompletableFuture<RaftClientReply> replyFuture = getClient().async()
-          .watch(index, RaftProtos.ReplicationLevel.ALL_COMMITTED);
+      CompletableFuture<RaftClientReply> replyFuture = getClient().async().watch(index, watchType);
       final RaftClientReply reply = replyFuture.get();
-      final long updated = updateCommitInfosMap(reply);
-      Preconditions.checkState(updated >= index);
-      return newWatchReply(index, ReplicationLevel.ALL_COMMITTED, updated);
+      if (!reply.isSuccess()) {
+        LOG.error("{} way commit failed", watchType, reply.getException());
+        throw reply.getException();
+      }
```

Comment:
I think we can remove this. I think `replyFuture#get` founds an exception, it will throw the exception automatically and the user code cannot access the `RaftClientReply`. It's one of the main reason of RATIS-2089.

Reply:
I see.